include/nodes/node.tcc

```diff
@@ -567,6 +567,94 @@ void mpm::Node<Tdim, Tdof, Tnphases>::apply_friction_constraints(double dt) {
   }
 }
 
+// !Apply absorbing constraints
+template <unsigned Tdim, unsigned Tdof, unsigned Tnphases>
+bool mpm::Node<Tdim, Tdof, Tnphases>::apply_absorbing_constraint(
+    unsigned dir, double delta, double h_min, double a, double b,
+    mpm::Position position) {
```

Comment:
@jgiven100 This doesn't work right?
```suggestion
template <unsigned Tdof, unsigned Tnphases>
bool mpm::Node<(2 or 3), Tdof, Tnphases>::apply_absorbing_constraint(
    unsigned dir, double delta, double h_min, double a, double b,
    mpm::Position position) {
```

Reply:
```
// !Apply absorbing constraints
template <unsigned Tdof, unsigned Tnphases>
bool mpm::Node<2, Tdof, Tnphases>::apply_absorbing_constraint(
    unsigned dir, double delta, double h_min, double a, double b,
    mpm::Position position) {
```

First compiler error
```
invalid use of incomplete type ‘class mpm::Node<2, Tdof, Tnphases>’
```
first SO hit -> https://stackoverflow.com/questions/165101/invalid-use-of-incomplete-type-error-with-partial-template-specialization
> You can't partially specialize a function. If you wish to do so on a member function, you must partially specialize the entire template (yes, it's irritating).